yarn-project/acir-simulator/src/client/pick_notes.ts

```diff
@@ -13,6 +13,16 @@ export enum SortOrder {
  * Options for selecting items from the database.
  */
 interface GetOptions {
+  /**
+   * An array of indices of the fields to select.
+   * Default: empty array.
+   */
+  selectBy?: number[];
```

Comment:
I'm finding the names of some of these to be confusing. Do the names follow some db lookup conventions that maybe I'm not familiar with?

Reply:
Do the `selectBy` and `selectValues` arrays have to be populated 'as one'? As in, do they have to be the same length, and does each item in `selectBy` correspond to the same position of `selectValues`?

Do the sortBy and sortOrder arrays have to be populated 'as one'? As in, do they have to be the same length, and does each item in `sortBy` correspond to the same position of `sortOrder`?

(I still don't think I can follow how I'm meant to specify these parameters)

Reply:
It's used for picking notes for noir contracts. So the options have been flatten like these because they are passed in via oracle calls. But I get that it can be confusing looking at this util by itself. I've changed it to convert them back to objects and use interfaces to make it clearer to read!